File: c++/Easy/Part002/Palindrome Missing Alphabet.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  string inputString;
  getline(cin, inputString); // Get input from the user

  vector<char> charList(inputString.begin(), inputString.end()); // Convert string to vector of characters
  int j = charList.size() - 1; // Initialize j to the last index of the vector

  for (int i = 0; i < charList.size(); i++) {
    if (charList[i] != charList[j]) {
      if (charList[i] == charList[j - 1] && i != j - 1) {
        cout << charList[j] << endl; // Print the character at index j
        break;
      } else {
        cout << charList[i] << endl; // Print the character at index i
        break;
      }
    }
    j--; // Decrement j
  }

  return 0;
}

File: Java/Easy/Part002/Palindrome Missing Alphabet.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String inputString = scanner.nextLine(); // Get input from the user

        List<Character> charList = new ArrayList<>();
        for (char c : inputString.toCharArray()) {
            charList.add(c);
        }

        int j = charList.size() - 1; // Initialize j to the last index of the list

        for (int i = 0; i < charList.size(); i++) {
            if (!charList.get(i).equals(charList.get(j))) {
                if (charList.get(i).equals(charList.get(j - 1)) && i != j - 1) {
                    System.out.println(charList.get(j)); // Print the character at index j
                    break;
                } else {
                    System.out.println(charList.get(i)); // Print the character at index i
                    break;
                }
            }
            j--; // Decrement j
        }

        scanner.close();
    }
}
